### PR TITLE
Fix inspections violations.

### PR DESCRIPTION
**Pull request by motlin**
Merely refreshing the TeamCity settings caused it to either start working again, or use newer versions of inspections, or something like that. A whole bunch of real violations started showing up against master.

### CODE REVIEW

File: eclipse-collections/src/main/java/org/eclipse/collections/impl/collection/mutable/AbstractMultiReaderMutableCollection.java [hunk 1]
```diff
@@ -77,10 +77,11 @@
  * AbstractMultiReaderMutableCollection is a common abstraction that provides thread-safe collection behaviors.
  * Subclasses of this class must provide implementations of getDelegate() and getLock().
  */
+@SuppressWarnings("unused")
 public abstract class AbstractMultiReaderMutableCollection<T> implements MutableCollection<T>
 {
-    protected transient ReadWriteLock lock;
-    protected transient ReadWriteLockWrapper lockWrapper;
+    protected ReadWriteLock lock;
```
Comment (donraab) [inline eclipse-collections/src/main/java/org/eclipse/collections/impl/collection/mutable/AbstractMultiReaderMutableCollection.java:83]: Why is it ok to remove transient here?

File: eclipse-collections/src/main/java/org/eclipse/collections/impl/collection/mutable/AbstractMultiReaderMutableCollection.java [hunk 2]
```diff
@@ -1981,13 +1982,11 @@ public void appendString(Appendable appendable, String start, String separator,
 
     public static class ReadWriteLockWrapper
     {
-        private final ReadWriteLock readWriteLock;
```
Comment (donraab) [inline eclipse-collections/src/main/java/org/eclipse/collections/impl/collection/mutable/AbstractMultiReaderMutableCollection.java:1984]: Is readWriteLock is not used?

File: eclipse-collections/src/main/java/org/eclipse/collections/impl/collection/mutable/AbstractMultiReaderMutableCollection.java (hunk 1 above)
Comment (motlin) [reply eclipse-collections/src/main/java/org/eclipse/collections/impl/collection/mutable/AbstractMultiReaderMutableCollection.java:83]: `transient` only affects `Serializable` classes. `AbstractMultiReaderMutableCollection` and its subclasses are not `Serializable`. Its subclasses are `Externalizable`

In theory, someone could write a new serializable subclass of AbstractMultiReaderMutableCollection. If you would prefer, I could bring back the keyword and suppress the warning.

File: eclipse-collections/src/main/java/org/eclipse/collections/impl/collection/mutable/AbstractMultiReaderMutableCollection.java (hunk 2 above)
Comment (motlin) [reply eclipse-collections/src/main/java/org/eclipse/collections/impl/collection/mutable/AbstractMultiReaderMutableCollection.java:1984]: Correct, the field is not used. The parameter is still used in the constructor.

```java
    public static class ReadWriteLockWrapper
    {
        private final ReadWriteLock readWriteLock;
        private final LockWrapper readLock;
        private final LockWrapper writeLock;

        public ReadWriteLockWrapper(ReadWriteLock readWriteLock)
        {
            this.readWriteLock = readWriteLock;
            this.readLock = new LockWrapper(readWriteLock.readLock());
            this.writeLock = new LockWrapper(readWriteLock.writeLock());
        }
...
```

File: eclipse-collections/src/main/java/org/eclipse/collections/impl/collection/mutable/AbstractMultiReaderMutableCollection.java (hunk 1 above)
Comment (donraab) [reply eclipse-collections/src/main/java/org/eclipse/collections/impl/collection/mutable/AbstractMultiReaderMutableCollection.java:83]: Yes, bring back the keyword and suppress the warning.

File: eclipse-collections/src/main/java/org/eclipse/collections/impl/collection/mutable/AbstractMultiReaderMutableCollection.java (hunk 2 above)
Comment (donraab) [reply eclipse-collections/src/main/java/org/eclipse/collections/impl/collection/mutable/AbstractMultiReaderMutableCollection.java:1984]: Ok

**Comment by motlin** [comment]
@donraab done

**Pull request merged**